pages/api/situation/index.js

```diff
@@ -0,0 +1,33 @@
+import cookie from "cookie";
+
+export default (req, res) => {
+  if (req.method === "POST") {
+    // update the user situation cookie
+    res.setHeader(
+      "Set-Cookie",
+      cookie.serialize("situation", req.body.situation, {
+        httpOnly: true,
+        secure: process.env.NODE_ENV !== "development",
+        maxAge: 60 * 60,
+        sameSite: "strict",
+        path: "/",
+      })
+    );
+  }
+
+  if (req.method === "DELETE") {
+    // set the user situation cookie to expire
+    res.setHeader(
+      "Set-Cookie",
+      cookie.serialize("situation", req.body.situation, {
+        httpOnly: true,
+        secure: process.env.NODE_ENV !== "development",
+        expires: new Date(0),
+        sameSite: "strict",
+        path: "/",
+      })
+    );
+  }
+
+  res.status(200).json({ success: true });
+};
```

Comment:
Does this need to be an index page? its intention might be cleared if it was simply /api/situation.js

Reply:
I would argue that it's pretty clear on what this is. Index pages are loaded my default if nothing specific is called, so a route to `/api/situation` will default to the index.

I originally created the folder thinking of future proofing in case the situation api would need more endpoints. It doesn't really look necessary now though.